### PR TITLE
Truncate value before cast to pass range-check assertions

### PR DESCRIPTION
**Pull request by bartwe**

### CODE REVIEW

**Comment by flibitijibibo** [comment]
This one's definitely for upstream - makes sense to me but not every platform may agree here...

File: src/SDL_utils.c [hunk 1]
```diff
@@ -123,7 +123,7 @@ static SDL_HashTable *SDL_objects;
 
 static Uint32 SDL_HashObject(const void *key, void *unused)
 {
-    return (Uint32)(uintptr_t)key;
+    return (Uint32)((uintptr_t)key & 0xffffffffu);
```
Comment (flibitijibibo) [inline src/SDL_utils.c:126]: Actually just to be _super_ annoying before we submit this upstream: Maybe we can use `SDL_UINT32_MAX` here just to make it clearer what this & is doing.

**Review by flibitijibibo — APPROVED** [review]
lgtm - let's send this to the upstream repo!

**Comment by bartwe** [comment]
How do i do that ?

**Comment by thatcosmonaut** [comment]
This is not specific to GPU and belongs in a PR upstream.

**Pull request closed without merging**